File: src/main/java/com/coffee/shop/api/demo/utils/HashMapConverter.java
package com.coffee.shop.api.demo.utils;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import java.io.IOException;
import java.util.Map;
import javax.persistence.AttributeConverter;
import lombok.extern.slf4j.Slf4j;

@Slf4j
public class HashMapConverter implements AttributeConverter<Map<String, Object>, String> {

    private final static ObjectMapper objectMapper = new ObjectMapper();

    @Override
    public String convertToDatabaseColumn(Map<String, Object> customerInfo) {

        String customerInfoJson = null;
        try {
            customerInfoJson = objectMapper.writeValueAsString(customerInfo);
        } catch (final JsonProcessingException e) {
            log.error("JSON writing error", e);
        }

        return customerInfoJson;
    }

    @Override
    public Map<String, Object> convertToEntityAttribute(String customerInfoJSON) {

        Map<String, Object> customerInfo = null;
        try {
            customerInfo = objectMapper.readValue(customerInfoJSON, new TypeReference<>() {
            });
        } catch (final IOException e) {
            log.error("JSON reading error", e);
        }
        return customerInfo;
    }

}

File: src/main/kotlin/com/coffee/shop/api/demo/utils/HashMapConverter.kt
package com.coffee.shop.api.demo.utils

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper
import lombok.extern.slf4j.Slf4j
import org.slf4j.LoggerFactory
import java.io.IOException
import javax.persistence.AttributeConverter

@Slf4j
class HashMapConverter : AttributeConverter<Map<String?, Any?>?, String?> {
    override fun convertToDatabaseColumn(customerInfo: Map<String?, Any?>?): String? {
        var customerInfoJson: String? = null
        try {
            customerInfoJson = objectMapper.writeValueAsString(customerInfo)
        } catch (e: JsonProcessingException) {
            log.error("JSON writing error", e)
        }
        return customerInfoJson
    }

    override fun convertToEntityAttribute(customerInfoJSON: String?): Map<String?, Any?>? {
        var customerInfo: Map<String?, Any?>? = null
        try {
            customerInfo = objectMapper.readValue(
                customerInfoJSON,
                object : TypeReference<Map<String?, Any?>?>() {})
        } catch (e: IOException) {
            log.error("JSON reading error", e)
        }
        return customerInfo
    }

    companion object {
        private val objectMapper = ObjectMapper()
        private val log = LoggerFactory.getLogger(Companion::class.java)
    }
}